src/pages/BlockDetail/index.tsx

```diff
@@ -48,12 +49,17 @@ export default () => {
     },
   )
 
+  const blockHeight = Number(blockHeightOrHash)
   const pageSize = queryBlockTransactions.data?.pageSize ?? pageSizeParam
 
   return (
     <Content>
       <BlockDetailPanel className="container">
-        <BlockOverviewCard blockHeightOrHash={blockHash ?? blockHeightOrHash} block={block} />
+        <BlockOverviewCard
+          blockHeightOrHash={blockHash ?? blockHeightOrHash}
+          block={block}
+          blockNumber={isTransactionHash(blockHeightOrHash) || Number.isNaN(blockHeight) ? block.number : blockHeight}
```

Comment:
With this implementation, it is still possible to get `defaultBlock.number (0)` (when the page is loaded and the block request has not yet received a response).

Therefore, my suggestion is to use the implementation of `getNextBlockHeight` to explicitly indicate the possibility of BlockHeight being `null`.

Reply:
I didn't use the default block data at any time, I think it is not possible.

Reply:
> I didn't use the default block data at any time, I think it is not possible.

The original cause of this bug is due to the usage of defaultBlock.

Code location: https://github.com/Magickbase/ckb-explorer-frontend/blob/a5b5c67d238f54feca539c15577b75e6511ee04c/src/pages/BlockDetail/index.tsx#L22C36-L22C52

Reply:
> > I didn't use the default block data at any time, I think it is not possible.
> 
> The original cause of this bug is due to the usage of defaultBlock.
> 
> Code location: https://github.com/Magickbase/ckb-explorer-frontend/blob/a5b5c67d238f54feca539c15577b75e6511ee04c/src/pages/BlockDetail/index.tsx#L22C36-L22C52

But I didn't use block number in default block data in new code, only when the `blockHeightOrHash ` is hash or undefined

Reply:
> But I didn't use block number in default block data in new code, only when the `blockHeightOrHash ` is hash or undefined

```
const block = queryBlock.data ?? defaultBlockInfo
...
blockNumber={isTransactionHash(blockHeightOrHash) || Number.isNaN(blockHeight) ? block.number : blockHeight}
```

Reply:
> > But I didn't use block number in default block data in new code, only when the `blockHeightOrHash ` is hash or undefined
> 
> ```
> const block = queryBlock.data ?? defaultBlockInfo
> ...
> blockNumber={isTransactionHash(blockHeightOrHash) || Number.isNaN(blockHeight) ? block.number : blockHeight}
> ```

When the `blockHeightOrHash` is `NaN` or block hash, the default value should be shown I think. Like https://explorer.nervos.org/block/0x9495fb64e32820c4d26c204a62d150679c51a4d5e96ff9c5e20e0543c501b8cf, do you know the block number when the page loaded?

Reply:
> When the `blockHeightOrHash` is `NaN` or block hash, the default value should be shown I think. Like https://explorer.nervos.org/block/0x9495fb64e32820c4d26c204a62d150679c51a4d5e96ff9c5e20e0543c501b8cf, do you know the block number when the page loaded?

The current issue is that this implementation can have bugs in certain situations. There are many other methods available to display default values, and the solution I initially suggested can meet your requirements. There is no need to use an approach that has bugs.

We should strive to make the code work in various scenarios as much as possible, rather than encountering exceptional behavior in rare branches.

```tsx
const blockHeight = isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number
const getNextBlockHeight = blockHeight == null ? null : ((step: number) => blockHeight + step)

BO = <BlockOverview getNextBlockHeight={getNextBlockHeight} block={block} {...otherProps} />

Prev = <Link to={`/block/${getNextBlockHeight?.(-1) ?? 0}`} disabled={!getNextBlockHeight} />
Next = <Link to={`/block/${getNextBlockHeight?.(1) ?? 0}`} disabled={!getNextBlockHeight} />
```

In this solution, you can see that the `block` is still being passed normally to `BlockOverview`, but the `getNextBlockHeight` is used in the `Link` to prevent potential bugs in various scenarios.

If you want to display default values, you can continue using `block` in `BlockOverview`. You can even adjust the `block` property of `BlockOverview` to be nullable, allowing you to handle default values internally using `defaultBlockInfo` or `blockNumber: block.number ?? 0`, or any other approach you prefer.

Reply:
> > When the `blockHeightOrHash` is `NaN` or block hash, the default value should be shown I think. Like https://explorer.nervos.org/block/0x9495fb64e32820c4d26c204a62d150679c51a4d5e96ff9c5e20e0543c501b8cf, do you know the block number when the page loaded?
> 
> The current issue is that this implementation can have bugs in certain situations. There are many other methods available to display default values, and the solution I initially suggested can meet your requirements. There is no need to use an approach that has bugs.
> 
> We should strive to make the code work in various scenarios as much as possible, rather than encountering exceptional behavior in rare branches.
> 
> ```tsx
> const blockHeight = isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number
> const getNextBlockHeight = blockHeight == null ? null : ((step: number) => blockHeight + step)
> 
> BO = <BlockOverview getNextBlockHeight={getNextBlockHeight} block={block} {...otherProps} />
> 
> Prev = <Link to={`/block/${getNextBlockHeight?.(-1) ?? 0}`} disabled={!getNextBlockHeight} />
> Next = <Link to={`/block/${getNextBlockHeight?.(1) ?? 0}`} disabled={!getNextBlockHeight} />
> ```
> 
> In this solution, you can see that the `block` is still being passed normally to `BlockOverview`, but the `getNextBlockHeight` is used in the `Link` to prevent potential bugs in various scenarios.
> 
> If you want to display default values, you can continue using `block` in `BlockOverview`. You can even adjust the `block` property of `BlockOverview` to be nullable, allowing you to handle default values internally using `defaultBlockInfo` or `blockNumber: block.number ?? 0`, or any other approach you prefer.

I think the same thing we did.
1. If the page opened with https://explorer.nervos.org/block/n, default block number 0 will be displayed.
2. If the page opened with block height, the block number will be displayed, and the next or previous page will also display the number.
3. If the page opened with transaction hash, the block number 0 will be displayed until the data fetched.

And the 0 will not be displayed when users switch the block quickly. 

I want to know what other bugs are. Or you just want to say it is better to code as you did, I would change the code style the same as you.

Reply:
> I want to know what other bugs are.

When the page is loaded and block requests have not yet received a response, switching blocks can result in unexpected next blocks.

This is more likely to occur in weak network environments or when the server is under high load and the response is slow.

https://github.com/Magickbase/ckb-explorer-frontend/assets/9160743/c8f62629-b2ac-42da-8fe8-0241adce1dbf

> Or you just want to say it is better to code as you did, I would change the code style the same as you.

It's not a matter of coding style. It's just about changing to a solution that doesn't rely on defaultBlockInfo to calculate the next block. If you want to stick with the current coding style, that's perfectly fine.

```tsx
const BlockOverview: FC<{ blockNumber?: number }>;
const BO = <BlockOverview blockNumber={isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number} />
```

Reply:
> > I want to know what other bugs are.
> 
> When the page is loaded and block requests have not yet received a response, switching blocks can result in unexpected next blocks.
> 
> This is more likely to occur in weak network environments or when the server is under high load and the response is slow.
> 
>  Untitled.video.1.mp4 
> > Or you just want to say it is better to code as you did, I would change the code style the same as you.
> 
> It's not a matter of coding style. It's just about changing to a solution that doesn't rely on defaultBlockInfo to calculate the next block. If you want to stick with the current coding style, that's perfectly fine.
> 
> ```tsx
> const BlockOverview: FC<{ blockNumber?: number }>;
> const BO = <BlockOverview blockNumber={isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number} />
> ```

I did not use the defaultBlockInfo to calculate the next block number. Only 0 (default block number) could be displayed when using hash to load the page, however I think in this case, the solution should be to set the button disabled. I think I did the same thing as you said.

```
const BlockOverview: FC<{ blockNumber?: number }>;
const BO = <BlockOverview blockNumber={isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number} />
```

The block number is also 0 until the data returned.
I will change the code to what you just suggest, but I still think we do the same thing, and represent the same result.

Reply:
> I did not use the defaultBlockInfo to calculate the next block number

However, in reality, the code logic in the current PR operates like this in extreme scenarios:

```tsx
const block = queryBlock.data ?? defaultBlockInfo
...
blockNumber={isTransactionHash(blockHeightOrHash) || Number.isNaN(blockHeight) ? block.number : blockHeight}
...
<Link to={`/block/${+blockNumber + 1}`}
```

Think about it carefully, and you can also see that I reproduced this issue in the video.

> the solution should be to set the button disabled.

This is the issue I need you to address. Apart from the difference in code style, I believe this is the only distinction between our solutions.

>I will change the code to what you just suggest, but I still think we do the same thing, and represent the same result.

In fact, you did not disable the button; otherwise, I would not be able to reproduce this issue. I think you may need to review our entire conversation starting from the initial comments. I have been emphasizing this issue from the beginning.

Reply:
> I think I did the same thing as you said.

> > Or you just want to say it is better to code as you did, I would change the code style the same as you.
> 
> It's not a matter of coding style. It's just about changing to a solution that doesn't rely on defaultBlockInfo to calculate the next block. If you want to stick with the current coding style, that's perfectly fine.
> 
> ```tsx
> const BlockOverview: FC<{ blockNumber?: number }>;
> const BO = <BlockOverview blockNumber={isBlockHeight(blockHeightOrHash) ? blockHeightOrHash : queryBlock.data?.number />
> ```

Note that the code snippet I provided earlier was just to show you one possible solution to implement link disabling in extreme scenarios while maintaining your code style. It was not my original complete solution. You might have misunderstood it as my new solution, which is why you thought it was the same thing.